neo/rawio/neuralynxrawio.py

```diff
@@ -531,171 +588,276 @@ def read_ncs_files(self, ncs_filenames):
-    else:
-        assert 'ApplicationName' in info
-        pattern = r'(\S*) "([\S ]*)"'
-        match = re.findall(pattern, info['ApplicationName'])
-        assert len(match) == 1, 'impossible to find application name and version'
-        info['ApplicationName'], app_version = match[0]
-    info['ApplicationVersion'] = distutils.version.LooseVersion(app_version)
-
-    # convert bit_to_microvolt
-    if 'bit_to_microVolt' in info:
-        btm_entries = re.findall(r'\S+', info['bit_to_microVolt'])
-        if len(btm_entries) == 1:
-            btm_entries = btm_entries * len(info['channel_ids'])
-        info['bit_to_microVolt'] = [float(e) * 1e6 for e in btm_entries]
-        assert len(info['bit_to_microVolt']) == len(info['channel_ids']), \
-            'Number of channel ids does not match bit_to_microVolt conversion factors.'
-
-    if 'InputRange' in info:
-        ir_entries = re.findall(r'\w+', info['InputRange'])
-        if len(ir_entries) == 1:
-            info['InputRange'] = [int(ir_entries[0])] * len(chid_entries)
+                info['InputRange'] = [int(e) for e in ir_entries]
+            assert len(info['InputRange']) == len(chid_entries), \
+                'Number of channel ids does not match input range values.'
+
+        # Filename and datetime depend on app name, app version, and -FileVersion
+        an = info['ApplicationName']
+        if an == 'Cheetah':
+            av = info['ApplicationVersion']
+            if av < '5':
+                hpd = NlxHeader.header_pattern_dicts['bv5']
+            elif av <= '5.6.4':
+                hpd = NlxHeader.header_pattern_dicts['bv5.6.4']
+            else:
+                hpd = NlxHeader.header_pattern_dicts['def']
+        elif an == 'BML':
+            hpd = NlxHeader.header_pattern_dicts['bv5']
         else:
-            info['InputRange'] = [int(e) for e in ir_entries]
-        assert len(info['InputRange']) == len(chid_entries), \
-            'Number of channel ids does not match input range values.'
-
-    # filename and datetime depend on app name and its version
-    if info['ApplicationName'] == 'Cheetah':
-        if info['ApplicationVersion'] <= '5.6.4':
-            old_date_format = True
+            hpd = NlxHeader.header_pattern_dicts['def']
+
+        original_filename = re.search(hpd['filename_regex'], txt_header).groupdict()['filename']
+
+        # opening time
+        dt1 = re.search(hpd['datetime1_regex'], txt_header).groupdict()
+        info['recording_opened'] = datetime.datetime.strptime(
+            dt1['date'] + ' ' + dt1['time'], hpd['datetimeformat'])
+
+        # close time, if available
+        if 'datetime2_regex' in hpd:
+            dt2 = re.search(hpd['datetime2_regex'], txt_header).groupdict()
+            info['recording_closed'] = datetime.datetime.strptime(
+                dt2['date'] + ' ' + dt2['time'], hpd['datetimeformat'])
+
+        return info
+
+    def typeOfRecording(self):
```

Comment:
This function name should be lower case and with underscores for compatibility with PEP8 and consistency with other functions in Neo.